teraserver/python/tests/modules/FlaskModule/API/user/test_UserQueryDeviceTypes.py

```diff
@@ -0,0 +1,183 @@
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device_2'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # update the key to key already created - Fail expected
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 8,
+                                  'device_type_key': 'new_device'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 500)
+
+        # update the key to key already created - pass expected
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 8,
+                                  'device_type_key': 'new_device_3'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # update the name - Pass expexted
+        params = {'device_type': {'device_type_name': 'NEW_DEVICE_TYPE',
+                                  'id_device_type': 8,
+                                  'device_type_key': 'new_device_3'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # Delete the object created by the test
+        params = {'device_type_key': 'new_device'}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        params = {'device_type_key': 'new_device_3'}
+        response = self._delete_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+    def test_delete_as_admin(self):
+        # This test should be run on a clean server (i.e. with only the defaults created)
+        # Else, the ID of the new devices created wont be aligned and the tests will fail
+        # Create a new device type ID = 5
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
+        # ID = 6
+        params = {'device_type': {'device_type_name': 'New_Device_Type',
+                                  'id_device_type': 0,
+                                  'device_type_key': 'new_device_1'}}
+        response = self._post_with_http_auth(username='admin', password='admin', payload=params)
+        self.assertEqual(response.status_code, 200)
+
```

Comment:
Instead of supposing returned IDs values, keep the values in a variable. For example:
current_id = json_data['id_device_type']

Otherwise, this will cause problems if running the tests multiple times without dropping the database (and thus, the ids will never have the right values after the first time the test is ran).